From 3ec5f9488258d387d0b6a523012eed35a1c81684 Mon Sep 17 00:00:00 2001
From: WangYuli <wangyuli@uniontech.com>
Date: Tue, 7 May 2024 08:27:31 +0800
Subject: [PATCH 004/360] deepin/defconfig: Enable werror for kernel builds

Enable this option can prevent from ignoring warnings which
resulting in problems that difficult to locate and solve.

Also, it encourages to program more regularly and carefully,
avoiding potential problems as much as possible.

So enable it on arm64, loongarch and x86.

Signed-off-by: WangYuli <wangyuli@uniontech.com>
---
 arch/arm64/configs/deepin_arm64_desktop_defconfig        | 1 -
 .../loongarch/configs/deepin_loongarch_desktop_defconfig | 4 ++--
 arch/x86/configs/deepin_x86_desktop_defconfig            | 9 ++++-----
 3 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/configs/deepin_arm64_desktop_defconfig b/arch/arm64/configs/deepin_arm64_desktop_defconfig
index b2e9134a4542..0df23bbc6009 100644
--- a/arch/arm64/configs/deepin_arm64_desktop_defconfig
+++ b/arch/arm64/configs/deepin_arm64_desktop_defconfig
@@ -1,5 +1,4 @@
 CONFIG_COMPILE_TEST=y
-# CONFIG_WERROR is not set
 CONFIG_BUILD_SALT="arm64-desktop-hwe"
 CONFIG_KERNEL_ZSTD=y
 CONFIG_DEFAULT_HOSTNAME=""
diff --git a/arch/loongarch/configs/deepin_loongarch_desktop_defconfig b/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
index 68b6ce03077a..50bffac5704d 100644
--- a/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
+++ b/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
@@ -1,7 +1,7 @@
 CONFIG_COMPILE_TEST=y
-# CONFIG_WERROR is not set
-CONFIG_LOCALVERSION="-loongarch-desktop-hwe"
+CONFIG_BUILD_SALT="-loongarch-desktop-hwe"
 CONFIG_KERNEL_ZSTD=y
+CONFIG_DEFAULT_HOSTNAME=""
 CONFIG_SYSVIPC=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_WATCH_QUEUE=y
diff --git a/arch/x86/configs/deepin_x86_desktop_defconfig b/arch/x86/configs/deepin_x86_desktop_defconfig
index d9b2a68b912d..80a96238c822 100644
--- a/arch/x86/configs/deepin_x86_desktop_defconfig
+++ b/arch/x86/configs/deepin_x86_desktop_defconfig
@@ -1,5 +1,5 @@
 CONFIG_COMPILE_TEST=y
-# CONFIG_WERROR is not set
+CONFIG_UAPI_HEADER_TEST=y
 CONFIG_BUILD_SALT="-amd64-desktop-hwe"
 CONFIG_KERNEL_ZSTD=y
 CONFIG_SYSVIPC=y
@@ -2652,7 +2652,6 @@ CONFIG_MFD_AAT2870_CORE=y
 CONFIG_MFD_BCM590XX=m
 CONFIG_MFD_BD9571MWV=m
 CONFIG_MFD_AXP20X_I2C=m
-CONFIG_MFD_CS42L43_SDW=m
 CONFIG_MFD_MADERA=m
 CONFIG_MFD_MADERA_I2C=m
 CONFIG_MFD_MADERA_SPI=m
@@ -3632,8 +3631,6 @@ CONFIG_SND_SOC_CS35L35=m
 CONFIG_SND_SOC_CS35L36=m
 CONFIG_SND_SOC_CS35L45_SPI=m
 CONFIG_SND_SOC_CS35L45_I2C=m
-CONFIG_SND_SOC_CS42L43=m
-CONFIG_SND_SOC_CS42L43_SDW=m
 CONFIG_SND_SOC_CS42L51_I2C=m
 CONFIG_SND_SOC_CS42L52=m
 CONFIG_SND_SOC_CS42L56=m
@@ -5575,7 +5572,9 @@ CONFIG_DEBUG_INFO_DWARF5=y
 CONFIG_DEBUG_INFO_BTF=y
 CONFIG_MODULE_ALLOW_BTF_MISMATCH=y
 CONFIG_GDB_SCRIPTS=y
-CONFIG_FRAME_WARN=1024
+CONFIG_FRAME_WARN=4096
+CONFIG_STRIP_ASM_SYMS=y
+CONFIG_READABLE_ASM=y
 CONFIG_HEADERS_INSTALL=y
 CONFIG_DEBUG_SECTION_MISMATCH=y
 CONFIG_STACK_VALIDATION=y
-- 
2.20.1

